src/server/parser/parser.ts

```diff
@@ -1854,6 +1924,8 @@ export class Parser
 		const token = this.lexer.token
 		if (config.allowExtStmt && token.typeIsOneOf(TokenType.visibility))
 			return this.parseVisibility()
+		if (config.isEnum)
```

Comment:
This should probably check `&& token.typeIsOneOf(TokenType.enumDef)`

Reply:
It comes down to: Do we want to only allow enums to contain enumeration values, or do we want to allow them to contain conditional logic, visibility statements, functions, operator overloads, etc?

One of the biggest problems C++ has always had with its enumerations (after fixing the scope problem) is there's no way to directly attach functionality to the type such as helpers to string-ify the enumeration values, and operators to help build complex values from enumerated values. If you agree that can make sense here, and given this current setup does allow visibility statements already, then we think checking for TokenType.enumDef is correct, otherwise we agree with the restriction